src/pkg/scan/sbom/sbom.go

```diff
@@ -164,3 +196,146 @@ func retrieveSBOMContent(rawReport string) ([]byte, *v1.Scanner, error) {
 	}
 	return sbomContent, rpt.Scanner, nil
 }
+
+func (h *scanHandler) MakeReportPlaceHolder(ctx context.Context, art *artifact.Artifact, r *scanner.Registration) (rps []*scanModel.Report, err error) {
+	var reports []*scanModel.Report
+	mgr := h.SBOMMgrFunc()
+	mimeTypes := r.GetProducesMimeTypes(art.ManifestMediaType, v1.ScanTypeSbom)
+	sbomReports, err := mgr.GetBy(h.cloneCtx(ctx), art.ID, r.UUID, mimeTypes)
+	if err != nil {
+		return nil, err
+	}
+	if err := h.deleteArtifactAccessories(ctx, sbomReports); err != nil {
+		return nil, err
+	}
+	for _, mt := range mimeTypes {
+		report := &sbom.Report{
+			ArtifactID:       art.ID,
+			RegistrationUUID: r.UUID,
+			MimeType:         mt,
+		}
+
+		create := func(ctx context.Context) error {
+			reportUUID, err := Mgr.Create(ctx, report)
+			if err != nil {
+				return err
+			}
+			report.UUID = reportUUID
+			return nil
+		}
+
+		if err := orm.WithTransaction(create)(orm.SetTransactionOpNameToContext(ctx, "tx-make-report-placeholder")); err != nil {
+			return nil, err
+		}
+		reports = append(reports, &scanModel.Report{
+			RegistrationUUID: r.UUID,
+			MimeType:         mt,
+			UUID:             report.UUID,
+		})
+	}
+
+	return reports, nil
+}
+
+// deleteArtifactAccessories delete the accessory in reports, only delete sbom accessory
+func (h *scanHandler) deleteArtifactAccessories(ctx context.Context, reports []*sbom.Report) error {
```

Comment:
If `// deleteArtifactAccessories delete the accessory in reports, only delete sbom accessory`, shall we rename the function to be something like `deleteArtifactSBOMAccessories`.

Reply:
Updated